Pulumi v3.62.0 fails to build from release tarball

### What happened?

Build Pulumi v3.62.0 from the release tarball on the releases page.

### Expected Behavior

Expected to get a `pulumi` binary.

### Steps to reproduce

```sh
pulumictlVersion='0.0.42'

chmod -R u+w GO
rm -rf GO

export GOPATH=$(pwd)/GO
rm -rf "${GOPATH}"
mkdir -p "${GOPATH}"

export PATH="/home/rkeene/devel/x/node/TMP/pulumi/go-1.18.1/bin:${PATH}"
export PATH="/home/rkeene/devel/x/node/TMP/pulumi/pulumictl-${pulumictlVersion}:${PATH}"
( cd sdk && go mod tidy && go mod download ) || exit 1
( cd pkg && go mod tidy && go mod download ) || exit 1

chmod -R u+w GO

echo '+ # disable network access'
enable -f /home/rkeene/devel/x/node/TMP/pulumi/bash-drop-network-access/dropnet.so dropnet

make brew
```

### Output of `pulumi about`

Not possible since the build failed.

### Additional context

Build fails with:

```text
+ cd pkg
+ go install -ldflags '-X github.com/pulumi/pulumi/pkg/v3/version.Version=0.1.0-alpha.1673497087+g4996550f' github.com/pulumi/pulumi/pkg/v3/cmd/pulumi
# github.com/pulumi/pulumi/pkg/v3/backend/filestate
backend/filestate/backend.go:121:24: undefined: atomic.Pointer
backend/filestate/backend.go:385:18: undefined: atomic.Int64
backend/filestate/backend.go:388:19: undefined: atomic.Int64
```

It has been indicated that CONTRIBUTING.md needs to be updated to specify that Go v1.19.something is now required.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -29,7 +29,7 @@ Finally, please limit your pull requests to contain only one feature at a time. 
 
 You'll want to install the following on your machine:
 
-- [Go 1.18 or later](https://go.dev/dl/)
+- [Go](https://go.dev/dl/) (a [supported version](https://go.dev/doc/devel/release#policy))
 - [NodeJS 14.X.X or later](https://nodejs.org/en/download/)
 - [Python 3.6 or later](https://www.python.org/downloads/)
 - [.NET](https://dotnet.microsoft.com/download)

